doc/computing/randomness.rst

```diff
@@ -0,0 +1,308 @@
+.. Places parent toc into the sidebar
+
+:parenttoc: True
+
+.. _randomness:
+
+Controlling randomness: good practices and common pitfalls
+==========================================================
+
+Some scikit-learn objects are inherently random. These are usually estimators
+(e.g. :class:`~sklearn.ensemble.RandomForestClassifier`) and cross-validation
+splitters (e.g. :class:`~sklearn.model_selection.KFold`). The randomness of
+these objects is controlled via their `random_state` parameter, as described
+in the :term:`Glossary <random_state>`. This section expands on the glossary
+entry, and describes good practice and common pitfalls w.r.t. to this
+subtle parameter.
+
+The `random_state` parameter determines whether multiple calls to :term:`fit`
+(for estimators) or to :term:`split` (for cv spliters) will produce the same
+results, according to these rules:
+
+- If an int is passed, calling `fit()` or `split()` multiple times yields the
+  same results.
+- If `None` or a `RandomState` instance is passed: `fit()` and `split()` will
+  yield different results each time they are called, producing an event chain
```

Comment:
We say in lines 69-76 that a `RandomState` instance would give us reproducible results across executions (with possible surprising and unwanted effects).

Maybe, to avoid confusion, we could omit `RandomState` instance here.

```suggestion
- If `None` is passed,`fit` and `split` will yield different
   results each time they are called, producing an event chain
```

Reply:
I don't think we want to omit instances here, the whole guide is about instances. Also the section below is about reproducible results across executions, which is a different thing from getting constant results across calls to fit / split